k-es-grpc/src/main/kotlin/no/ks/kes/grpc/GrpcEventSubscriberFactory.kt

```diff
@@ -19,25 +16,73 @@ import kotlin.reflect.KClass
 
 private val log = KotlinLogging.logger {}
 
+private const val MAX_RECONNECT_RETRIES = 10L
+
 class GrpcEventSubscriberFactory(
     private val eventStoreDBClient: EventStoreDBClient,
     private val serdes: EventSerdes,
     private val category: String,
     private val metadataSerdes: EventMetadataSerdes<out Metadata>? = null
 ) : EventSubscriberFactory<GrpcSubscriptionWrapper> {
 
+    private val retryCount = AtomicLong(0)
+    private var firstOnCancelled: Instant? = null
```

Comment:
Med denne staten i factory-klassen burde vel ikke createSubscriber kalles flere ganger på samme instans. Burde man forhindre det ved å kaste en exception om noen prøver på det?

Reply:
Endte med å flytte subscriber ned i wrapper klassen, sammen med logikk for reconnect. Å omdefinere factory endte ble et blindspor om vi ikke skulle skrive om hele k-es